src/mixer.cpp

```diff
@@ -77,18 +77,22 @@ int mixer_connect_input(mixer_t *mixer, float ampfactor, float balance) {
 	mixer->inputs[i].ready = false;
 	mixer->inputs[i].has_signal = false;
 	mixer->inputs[i].input_overrun_count = 0;
-	SET_BIT(mixer->input_mask, i);
-	SET_BIT(mixer->inputs_todo, i);
+	mixer->input_mask[i] = 1;
+	mixer->inputs_todo[i] = 1;
 	mixer->enabled = true;
 	debug_print("ampfactor=%.1f ampl=%.1f ampr=%.1f\n", mixer->inputs[i].ampfactor, mixer->inputs[i].ampl, mixer->inputs[i].ampr);
 	return(mixer->input_count++);
 }
 
 void mixer_disable_input(mixer_t *mixer, int input_idx) {
+    int mixer_sum = 0;
 	assert(mixer);
 	assert(input_idx < mixer->input_count);
-	RESET_BIT(mixer->input_mask, input_idx);
-	if(mixer->input_mask == 0) {
+	mixer->input_mask[input_idx] = 0;
+	for (int i = 0; i < MAX_MIXINPUTS; i++) {
```

Comment:
I think this can go up to `mixer->input_count` rather than `MAX_MIXINPUTS` (check other places as well)